### PR TITLE
Use .netstandard2.0 for default extension

### PR DESCRIPTION
**Pull request by twsouthwick**
We want to use .NET Standard for the extensions if possible so that we
could potentially run them in Visual Studio at some point. The reason
the default extension was not .NET Standard is because the try-convert
stuff uses MSBuild directly and was forcing .NET 5+ to be the TFM. This
change moves try-convert to be its own extension which allows the
default extension to be .NET Standard.

### CODE REVIEW

File: eng/DependencyValidation/DependencyValidation.modelproj [hunk 1]
```diff
@@ -34,6 +34,26 @@
     </Validate>
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\..\modules\try-convert\src\MSBuild.Abstractions\MSBuild.Abstractions.csproj">
```
Comment (mjrousos) [inline eng/DependencyValidation/DependencyValidation.modelproj:37]: This may become clearer later, but why are all these additional references needed now?

**Review by mjrousos — APPROVED** [review]
Good change; thanks!

File: eng/DependencyValidation/DependencyValidation.modelproj (hunk 1 above)
Comment (twsouthwick) [reply eng/DependencyValidation/DependencyValidation.modelproj:37]: Not really sure. It auto-added them and I'm not really sure how this project file works

Comment (twsouthwick) [reply eng/DependencyValidation/DependencyValidation.modelproj:37]: The modelproj format still uses the old style format, so......

Comment (mjrousos) [reply eng/DependencyValidation/DependencyValidation.modelproj:37]: Got it.

**Review by sunandabalu — APPROVED** [review]
What happens if we take any updates from try-convert, will any of these potentially change?

**Comment by twsouthwick** [comment]
@sunandabalu not that I know of. If it were to update to target .net 6, then we'd need to react to that. The nice thing about the extensions running in their own AssemblyLoadContext is that they are fairly isolated from each other.

**Pull request merged**